pageserver/src/bin/pageserver.rs

```diff
@@ -270,15 +270,31 @@ fn start_pageserver(
     WALRECEIVER_RUNTIME.block_on(pageserver::broker_client::init_broker_client(conf))?;
 
     // Initialize authentication for incoming connections
-    let auth = match &conf.auth_type {
-        AuthType::Trust => None,
-        AuthType::NeonJWT => {
-            // unwrap is ok because check is performed when creating config, so path is set and file exists
-            let key_path = conf.auth_validation_public_key_path.as_ref().unwrap();
-            Some(JwtAuth::from_key_path(key_path)?.into())
-        }
-    };
-    info!("Using auth: {:#?}", conf.auth_type);
+    let http_auth;
+    let pg_auth;
```

Comment:
why not initialize here with None and avoid the "else" clause?

Reply:
Another option to consider would be to assign them directly from the block expression: 

```rust
let (foo, bar) = { 
     // ...
}
```

Not sure which option would be nicer

Reply:
Probably this is another point for making it separate function